contents/linux/shell.md

```diff
@@ -0,0 +1,180 @@
+<frontmatter>
+  title: Introduction to Linux Shell
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Linux bash shell 
+
+Author: [Wang Junming](https://github.com/junming403)
+
+## Overview
+
+This chapter assumes that the reader is familiar with general concepts of operating systems. If you are not familiar with basic operating system concepts, I recommend you pick up the basic concepts first. The classic textbook of operating system **Mordern Operating Systems by Tanenbaum** is highly recommended. 
+
+<img src="images/shell-kernel.png" alt="shell-kernel" style="width:400px;"/>
+
+Computer can be divided into 3 abstraction layers, hardware, kernel and applications. Users cannot control the hardware directly, instructions have to be given through kernel as kernel is the one that controls the hardware. 
+Shell is the interface through which we can pass instructions to kernel, and then it could be executed on the hardware. In this chapter, we will look into **Linux command-line based shell(bash)**.
+
+> Linux is the best-known and most-used open source operating system. As an operating system, Linux is software that sits underneath all of the other software on a computer, receiving requests from those programs and relaying these requests to the computer’s hardware.
+
+[**bash**](https://www.gnu.org/software/bash/manual/html_node/) stands for **Bourne Again SHell**, an enhanced version of the original Unix Shell program. It has become the standard shell of various linux distributions. 
+
+## Why use command-line based shell?
```

Comment:
This section seems very technical! You could also add some points to convince total beginners to switch over to bash? People who have zero experience with command line may not be able to follow along with the code snippets you have put up.

Reply:
Actually I do assume some understanding of OS in order to understand this article, because OS is an essentially a must-learn for everyone of CS background.  I have listed points below for why should get started to using shell.
1. `Remote Management` 
2. `Stability`
3. `Freedom`
4. `Simplicity`